src/hoc/withTheme.js

```diff
@@ -23,25 +22,46 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
     static styledComponentId = 'withTheme'
 
     static contextTypes = {
-      [CHANNEL]: PropTypes.func,
+      [CHANNEL]: CONTEXT_CHANNEL_SHAPE,
```

Comment:
This is a hard breaking change that'll make every library built with a version of styled-components before this one incompatible, right?